src/main/java/org/opensearch/eval/samplers/TopNQuerySampler.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.eval.samplers;
+
+import org.opensearch.eval.model.ubi.query.UbiQuery;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * A sampler that selects the top N queries.
+ */
+public class TopNQuerySampler extends AbstractQuerySampler {
+
+    public static final String NAME = "topn";
+
+    final TopNQuerySamplerParameters parameters;
+
+    public TopNQuerySampler(final TopNQuerySamplerParameters parameters) {
+        this.parameters = parameters;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public Map<String, Long> sample(final Collection<UbiQuery> ubiQueries) {
+
+        final Map<String, Long> querySet = new HashMap<>();
+
+        // Remove duplicates from the queries.
+        final Set<UbiQuery> queriesWithoutDuplicates = new HashSet<>(ubiQueries);
+
+        // For getting the frequency of each user query.
+        final Map<String, Long> counts = ubiQueries.stream().collect(Collectors.groupingBy(UbiQuery::getUserQuery, Collectors.counting()));
+
+        if(parameters.getQuerySetSize() >= queriesWithoutDuplicates.size()) {
+
+            // Take all queries since the requested size is equal to or greater than the total number of queries.
+            for(final UbiQuery ubiQuery : queriesWithoutDuplicates) {
+                querySet.put(ubiQuery.getUserQuery(), counts.get(ubiQuery.getUserQuery()));
+            }
+
+        } else {
+
+            // Sort the queries by frequency, highest to lowest.
+            final TreeMap<String, Long> sortedMap = new TreeMap<>(Comparator.comparing(counts::get, Comparator.reverseOrder()));
```

Comment:
I'm unable to create query sets of a size greater than 4.
Specifying the size to 10, 25, 40 all give me the same four queries and these are not the most frequent ones.
Maybe we need to look into the code that retrieves all queries frequencies. Are we doing that with an aggregation or in another way?

Reply:
I'm now seeing we're using a scroll operation. 
I don't have an estimation whether aggregations or scroll operations are more efficient for our use case. Maybe it even depends on the total amount of data.
Aggregations can be heavy operations in our case as we need to group all the queries. Using scroll operations moves that complexity to the app. I'm probably more in favor of letting OpenSearch do the heavy lifting for us with aggregations. Less risk for the app running out of memory. What do you think?

Reply:
Yeah, that's a good point. Might be 6 of one and half dozen of the other. But, an aggregation would likely be simpler in the code. And, realistically, I don't think query sets will be made frequently enough to make the heavy operation on OpenSearch to be a problem.